portlayer/exec/container.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/metadata"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"golang.org/x/net/context"
+)
+
+var containers map[ID]*Container
+var containersLock sync.Mutex
+
+func init() {
+	containers = make(map[ID]*Container)
+}
+
+type State int
+
+const (
+	StateRunning = iota
+	StateStopped = iota
+)
+
+type Container struct {
+	sync.Mutex
+
+	ID         ID
+	ExecConfig *metadata.ExecutorConfig
+
+	vm *object.VirtualMachine
+}
+
+func NewContainer(id ID) *Handle {
+	con := &Container{
+		ID: id,
+	}
+
+	containersLock.Lock()
+	containers[id] = con
+	containersLock.Unlock()
+
+	return con.newHandle()
+}
+
+func GetContainer(id ID) *Handle {
+	containersLock.Lock()
+	defer containersLock.Unlock()
+
+	if con, ok := containers[id]; ok {
+		return con.newHandle()
+	}
+
+	return nil
+}
+
+func (c *Container) newHandle() *Handle {
+	return newHandle(c)
+}
+
+func (c *Container) CommitHandle(ctx context.Context, sess *session.Session, h *Handle) error {
```

Comment:
Why commit a handle instead of just passing the configspec in directly? This looks like we're blending the handle/cache code with the foundational PL functions.